2-mandatory/3-shopping-list.js

```diff
@@ -20,8 +20,16 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+   const resultObj = {};
+   resultObj.name = recipe.name;
+   resultObj.items = [];
+    
+   const pantryAllSet = new Set(Object.values(pantry).flat());
+    resultObj.items = [
+      ...new Set(recipe.ingredients.filter((x) => !pantryAllSet.has(x))),
+    ];
+return resultObj;
```

Comment:
good job @ShayanMahnam , This is another solution without using SET, well done :) 
`function createShoppingList(recipeObj){
  let result = {name: recipeObj.name , shoppingList: []}
  let allIHave = pantry.fridgeContents.concat(pantry.cupboardContents)
  recipeObj.ingredients.map(eachIng => {if(!allIHave.includes(eachIng)){
    result.shoppingList.push(eachIng)
  }})
   return result;
}`

Reply:
Nice one thanks. (added)